drivers/encx24j600/encx24j600.c

```diff
@@ -82,12 +82,19 @@ static inline void unlock(encx24j600_t *dev) {
 }
 
 void encx24j600_setup(encx24j600_t *dev, const encx24j600_params_t *params)
+{
+    encx24j600_setup_netdev(dev, params, 0);
+}
+
+void encx24j600_setup_netdev(encx24j600_t *dev, const encx24j600_params_t *params, uint8_t index)
```

Comment:
Hm, I'm not sure I agree with the function shuffling here.
The old `enxc24j600_setup()` now does more than before, and just calls the "new" function.
Wouldn't it make more sense to keep the `_setup()` as is and have the new `_setup_netdev()` call `_setup()` and `netdev_register()`?

Reply:
(I think the other drivers are changed in a similar way).

Reply:
> Hm, I'm not sure I agree with the function shuffling here. The old `enxc24j600_setup()` now does more than before, and just calls the "new" function. Wouldn't it make more sense to keep the `_setup()` as is and have the new `_setup_netdev()` call `_setup()` and `netdev_register()`?

I thought that we want all netdev drivers to register themselves (although I'm not 100% sure if this is the case), that's why I introduced this. It was a bit surprising not to get the registration callback in the app, so I had to go to the code to learn this was, in fact, not implemented.

Reply:
The change is valid and long overdue. In fact, I'm surprised we have not yet ported all netdevs to the new API. With this change it is now consistent with other netdevs and the vision we have for this subsystem.

The signature change was introduced to generate stable L2 addresses, which are derived from the CPU ID, the driver type, and the index in the params struct. Previously, disabling one netdev or changing the initialisation order resultet in different L2 addresses.

Maybe @benpicco can confirm that this change is well aligned with the direction we want the netdevs to evolve to.

Reply:
Yea I thought I added the register call to all drivers - looks like some have slipped through the cracks - thank you for handling this!

Reply:
OK!